golang/hyperbahn/client.go

```diff
@@ -119,9 +120,21 @@ func addPeer(ch *tchannel.Channel, hostPort string) {
 	peers.Add(hostPort)
 }
 
+func (c *Client) getServiceNames(otherServices []tchannel.Registrar) {
+	c.services = make([]string, 0, len(otherServices)+1)
+	c.services = append(c.services, c.tchan.PeerInfo().ServiceName)
+
+	for _, s := range otherServices {
+		c.services = append(c.services, s.ServiceName())
+	}
+}
+
 // Advertise advertises the service with Hyperbahn, and returns any errors on initial advertisement.
+// Advertise can register multiple services hosted on the same endpoint.
 // If the advertisement succeeds, a goroutine is started to re-advertise periodically.
-func (c *Client) Advertise() error {
+func (c *Client) Advertise(otherServices ...tchannel.Registrar) error {
```

Comment:
So, now if you have no subchannels, you still have to pass in the channel you used when you created the client, right?